dvc/remote/azure.py

```diff
@@ -65,14 +99,7 @@ def get_etag(self, path_info):
     def get_file_checksum(self, path_info):
         return self.get_etag(path_info)
 
-    def remove(self, path_info):
-        if path_info.scheme != self.scheme:
-            raise NotImplementedError
-
-        logger.debug(f"Removing {path_info}")
-        self.blob_service.delete_blob(path_info.bucket, path_info.path)
-
-    def _list_paths(self, bucket, prefix, progress_callback=None):
+    def list_paths(self, bucket, prefix, progress_callback=None):
```

Comment:
Isn't this `walk_files` ? :slightly_smiling_face:

Reply:
It probably should be. I think it makes sense to implement `walk_files` for the remotes which still do not have it as a part of this overall refactoring, but in this PR I was only looking to start moving functions into the tree classes without changing existing behavior and potentially breaking things.